### PR TITLE
added startAt and endAt props for video

### LINKED ISSUE #150: Some way to let the video start at a certain time
Discussed with @jonpacker on Discord, it would be cool to have a more obvious API for starting a video at 2 seconds:

way to video

### PR DESCRIPTION
**Pull request by Iamshankhadeep**
fixes #150

### CODE REVIEW

File: packages/core/src/video/VideoForDevelopment.tsx [hunk 1]
```diff
@@ -40,6 +32,7 @@ export const VideoForDevelopment: React.FC<RemotionVideoProps> = (props) => {
 
 		if (!playing || absoluteFrame === 0) {
 			videoRef.current.currentTime = currentTime;
+			videoRef.current.pause();
 		}
 
 		if (videoRef.current.paused && !videoRef.current.ended && playing) {
```
Comment (JonnyBurger) [inline packages/core/src/video/VideoForDevelopment.tsx]: I think the changes in this file are not necessary anymore right?

File: packages/core/src/video/Video.tsx [hunk 2]
```diff
@@ -1,11 +1,32 @@
...
-	return <VideoForRendering {...props} />;
+	return (
+		<Sequence
+			from={0 - startAtFrameNo}
+			durationInFrames={endAtFrameNO - startAtFrameNo}
+		>
+			<VideoForRendering {...props} />;
+		</Sequence>
+	);
 };
```
Comment (JonnyBurger) [inline packages/core/src/video/Video.tsx]: How about recursion? It would remove redudancy and be easier to reason about right?

```tsx
const {startAt, endAt, ...otherProps} = props
if (startAt || endAt) {
  return (
    <Sequence
      from={0 - startAtFrameNo}
	  durationInFrames={endAtFrameNO - startAtFrameNo}
    >
      <Video {...otherProps}>
    </Sequence>
  )
}
// Logic we had until now
```

File: packages/core/src/video/VideoForRendering.tsx [hunk 3]
```diff
@@ -20,7 +21,9 @@ export const VideoForRendering: React.FC<RemotionVideoProps> = ({
 		if (!videoRef.current) {
 			return;
 		}
-		const frameInSeconds = frame / videoConfig.fps;
+		const frameInSeconds = FEATURE_FLAG_V2_BREAKING_CHANGES
+			? (frame + 1) / videoConfig.fps
+			: frame / videoConfig.fps;
```
Comment (JonnyBurger) [inline packages/core/src/video/VideoForRendering.tsx]: Let's make this change in the audio branch instead, then we won't have conflicts

File: packages/docs/docs/video.md [hunk 4]
```diff
@@ -11,7 +11,7 @@ Videos with audio are not supported - yet! Audio will be muted in the final rend
 
 ## API / Example
 
-Use an import or require to load an audio file and pass it as the `src` prop. All the props that the native `<video>` element accepts (except `autoplay` and `controls`) will be forwarded (but of course not all are useful for Remotion). This means you can use all CSS to style the video.
+Use an import or require to load an video file and pass it as the `src` prop. All the props that the native `<video>` element accepts, `startAt` (optional) and `endAt` (optional) props (except `autoplay` and `controls`) will be forwarded (but of course not all are useful for Remotion). This means you can use all CSS to style the video.
```
Comment (JonnyBurger) [inline packages/docs/docs/video.md]: Nice! Makes it sound like `startAt` prop will be forwarded to the video element, which shouldn't happen. Can we leave this sentence and then mention `startAt` in a new paragraph? It will probably also be easier to read and digest for the reader.

**Review by JonnyBurger — CHANGES_REQUESTED** [review]

**Comment by JonnyBurger** [comment]
Thanks a lot! Super awesome stuff. I added some comments. When we are happy, we could also add it to the audio element!

File: packages/core/src/video/VideoForDevelopment.tsx (hunk 1 above)
Comment (Iamshankhadeep) [reply packages/core/src/video/VideoForDevelopment.tsx]: Yup changes in this file is not necessary.

File: packages/core/src/video/VideoForRendering.tsx (hunk 3 above)
Comment (Iamshankhadeep) [reply packages/core/src/video/VideoForRendering.tsx]: okay I'll revert this changes then.

File: packages/core/src/video/Video.tsx (hunk 2 above)
Comment (Iamshankhadeep) [reply packages/core/src/video/Video.tsx]: will surely do,

File: packages/docs/docs/video.md (hunk 4 above)
Comment (Iamshankhadeep) [reply packages/docs/docs/video.md]: done

**Review by JonnyBurger — CHANGES_REQUESTED** [review]
Looking good! Now we can refine.

Let's do the following to further increase stability and user experience:

- Remove the console.log
- Add the same behavior to <Audio/> component
- Add documentation for the props in audio component
- add some runtime validation if people pass in parameters that are not numbers (for non-typescript users), negative numbers, NaN, or if they pass `Infinity` as start numbers
- Add some tests to validate that errors are being thrown - you can probably copy paste it from `packages/core/src/test/sequence-validation.test.tsx` and adapt it to work for the Video component. Also add it to the Audio component
- Test the edge case: What if `endAt` is 0? It will lead to `durationInFrames={Infinity}`. this should probably be an error
- Also validate that endAt > startAt

I know it's a lot extra, but we can take our time to make it as robust as possible and help users if they are writing bugs!

**Comment by Iamshankhadeep** [comment]
> I know it's a lot extra, but we can take our time to make it as robust as possible and help users if they are writing bugs!

Done with everything. Have a look at it now. @JonnyBurger

**Review by JonnyBurger — APPROVED** [review]
